src/zkevm_specs/evm/execution_result/call.py

```diff
@@ -0,0 +1,156 @@
+from ...util import linear_combine, le_to_int, EMPTY_CODE_HASH
+from ..common_assert import assert_bool
+from ..step import Step
+from ..table import FixedTableTag, RWTableTag, CallContextTag
+from ..opcode import Opcode
+from .execution_result import ExecutionResult
+
+
+def call(curr: Step, next: Step, r: int, opcode: Opcode):
+    # Verify opcode
+    assert opcode == Opcode.CALL
+
+    # Verify depth
+    depth = curr.call_context_lookup(CallContextTag.Depth)
+    curr.fixed_lookup(FixedTableTag.Range1024, [depth - 1])
+
+    # Gas needs full decompression due to EIP 150
+    bytes_gas = curr.decompress(curr.stack_pop_lookup(), 32, r)
+    bytes_callee_address = curr.decompress(curr.stack_pop_lookup(), 32, r)
+    bytes_value = curr.decompress(curr.stack_pop_lookup(), 32, r)
+    bytes_cd_offset = curr.decompress(curr.stack_pop_lookup(), 32, r)
+    bytes_cd_length = curr.decompress(curr.stack_pop_lookup(), 5, r)
```

Comment:
Why limit the length to 5 bytes? I see in the Geth, the length uses Uint64 type. So it can be up to 8 bytes. Though it's impossible to have such large data, I feel it's better to be consistent?

Reply:
In EVM the `memory_address` could be up to <code>2<sup>37</sup>-32</code> in successful cases<sup>[1]</sup>. I think decompressing every `memory_address` into 5 bytes is reasonable, although it then could be up to <code>2<sup>40</sup>-1</code> which leads to inequivalent to current EVM when gas limit is raised to very large value <code>≈2<sup>55</sup></code>, but I don't think it's possible.

The reason `cd_offset` is decompressed into 32 bytes is because the behavior of EVM, which allows `offset` to be any value if the `length` is `0`.

[1]: https://github.com/ethereum/go-ethereum/blob/master/core/vm/gas_table.go#L33-L40